index.js

```diff
@@ -54,18 +68,204 @@ client.on("messageReactionAdd", async (reaction, user) => {
     }
 
     // Check if reaction emoji matches first target emoji:
-    if (reaction.emoji.name === firstTargetEmoji) {
-        emojiReactToSend(reaction.message, user, firstTargetChannelId);
+    if (reaction.emoji.name === PIN_TARGET_EMOJI) {
+        emojiReactToSend(reaction.message, user, PIN_TARGET_CHANNEL_ID);
     }
 
     // Check if reaction emoji matches second target emoji:
-    if (reaction.emoji.name === secondTargetEmoji) {
-        emojiReactToSend(reaction.message, user, secondTargetChannelId);
+    if (reaction.emoji.name === UNREAD_TARGET_EMOJI) {
+        emojiReactToSend(reaction.message, user, UNREAD_TARGET_CHANNEL_ID);
     }
 
     // Check if reaction emoji matches third target emoji:
-    if (reaction.emoji.name === thirdTargetEmoji) {
-        emojiReactToSend(reaction.message, user, thirdTargetChannelId);
+    if (reaction.emoji.name === TODO_TARGET_EMOJI) {
+        emojiReactToSend(reaction.message, user, TODO_TARGET_CHANNEL_ID);
+    }
+
+    // Check emojis for deleting todo:
+    const message = reaction.message.content;
+
+    if (reaction.emoji.name === DELETE_TODO_EMOJI) {
+        if (message.match(USERNAME_REGEX)[0] === `<@${user.id}>`) {
+            deleteTodo[0] = true;
+        }
+        if (message.match(USERNAME_REGEX)[1] === `<@${user.id}>`) {
+            deleteTodo[1] = true;
+        }
+        if (deleteTodo[0] === true && deleteTodo[1] === true) {
+            reaction.message.delete();
+            deleteTodo[0] = false;
+            deleteTodo[1] = false;
+        }
+    }
+});
+
+// Event listener for reactions removed from messages:
+client.on("messageReactionRemove", async (reaction, user) => {
+    // Ignore bot reactions:
+    if (user.bot) {
+        return;
+    }
+
+    // Fetch partials if necessary:
```

Comment:
Comment should be: Fetch full version if partial*
You are fetching the full info because Discord might not have given you all the info you needed (a partial)

Reply:
Good catch